crates/sui-bridge/src/sui_syncer.rs

```diff
@@ -1,14 +1,237 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-//! The Suisyncer module is responsible for synchronizing Events emitted on Sui blockchain from
-//! concerned bridge packages. Each package is associated with a cursor, and the syncer will
-//! only query from that cursor onwards. It's likely that SuiSyncer only tracks one package.
+//! The SuiSyncer module is responsible for synchronizing Events emitted on Sui blockchain from
+//! concerned bridge packages.
 
+use crate::{
+    error::BridgeResult,
+    retry_with_max_delay,
+    sui_client::{SuiClient, SuiClientInner},
+};
+use mysten_metrics::spawn_logged_monitored_task;
+use std::{collections::HashMap, sync::Arc};
+use sui_json_rpc_types::SuiEvent;
+use sui_types::{
+    base_types::ObjectID, digests::TransactionDigest, Identifier, SUI_SYSTEM_PACKAGE_ID,
+};
+use tokio::{
+    task::JoinHandle,
+    time::{self, Duration},
+};
+use tokio_retry::strategy::{jitter, ExponentialBackoff};
+use tokio_retry::Retry;
 
+// TODO: use the right package id
+const PACKAGE_ID: ObjectID = SUI_SYSTEM_PACKAGE_ID;
+const SUI_EVENTS_CHANNEL_SIZE: usize = 1000;
 
+/// Map from contract address to their start block.
+pub type SuiTargetModules = HashMap<Identifier, TransactionDigest>;
```

Comment:
For a new contract, what would be the TransactionDigest?

Reply:
it could be any transaction digest that happens before the new contract's existence - it's not very intuitive, i hope we have checkpoint as the cursor